qpmodel/stmt.cs

```diff
@@ -131,6 +131,13 @@ public virtual List<Row> Exec()
             {
                 Console.WriteLine(physicPlan_.Explain(queryOpt_.explain_));
             }
+
+            var dcontext = context as DistributedContext;
+            if (dcontext != null)
+            {
+                dcontext.machines_.WaitThreads();
```

Comment:
why we need to waitThreads() here?

Reply:
If the query plan(in LISP) as (PhysicGather(...(PhysicGather))), we must wait all threads to guarantee loop count of second PhysicGather is fixed value.